### PR TITLE
Offline sync settings UI

### PR DESCRIPTION
**Pull request by vargaat**
refs: MBL-16665
affects: Student
release note: none

test plan: See Figma designs in ticket.

## Screenshots

<table>
<tr>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [ ] Follow-up e2e test ticket created
- [x] A11y checked
- [x] Tested on phone
- [ ] Tested on tablet
- [ ] Tested in dark mode
- [ ] Tested in light mode
- [x] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>One or more files are below the minimum test coverage 50%</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-16738](https://instructure.atlassian.net/browse/MBL-16738)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.91% | 91.92% | -0.01%
Core/Core/Extensions/PublisherExtensions.swift | 22.73% | 22.73% | 0%
Core/Core/Courses/CourseDetails/ViewModel/CourseDetailsViewModel.swift | 48.89% | 48.89% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 013c056cb7a81b20e53e7ddf38d5d5df42e51781
</p>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

File: Core/Core/CourseSync/CourseSyncSettings/ViewModel/CourseSyncSettingsViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            .filter { !$0 }
+            .map { _ in true }
+            .handleEvents(receiveOutput: { [unowned self] _ in
+                turnOnWifiOnlySyncWhenUserCancelsConfirmation()
+            })
+            .assign(to: &$isShowingConfirmationDialog)
+    }
+
+    private func turnOnWifiOnlySyncWhenUserCancelsConfirmation() {
+        let hapticGenerator = UIImpactFeedbackGenerator(style: .light)
```
Comment (szabinst) [inline Core/Core/CourseSync/CourseSyncSettings/ViewModel/CourseSyncSettingsViewModel.swift:78]: 👍

**Review by szabinst — APPROVED** [review]
QA + 1

**Review by idevainstr — APPROVED** [review]
QA +1

**Pull request merged**